### PR TITLE
Move MockAppConfigurer into test source.

### PR DESCRIPTION
**Pull request by chtyim**
- It shouldn’t be in main.
- Make cdap-integration-test depends on app-fabric test-jar

### CODE REVIEW

File: cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java [hunk 1]
```diff
@@ -125,7 +126,7 @@ public ApplicationManager deployApplication(Id.Namespace namespace, Class<? exte
       if (configObject != null) {
         appConfig = GSON.toJson(configObject);
       } else {
-        configObject = (Config) configToken.getRawType().newInstance();
```
Comment (chtyim) [inline cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java]: IntelliJ complains about this.

File: cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java [hunk 2]
```diff
@@ -108,6 +108,7 @@ public ApplicationManager deployApplication(Id.Namespace namespace,
   }
 
   @Override
+  @SuppressWarnings("unchecked")
```
Comment (bdmogal) [inline cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java]: is this needed even after the changes below?

**Comment by bdmogal** [comment]
LGTM

File: cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java (hunk 2 above)
Comment (chtyim) [inline cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java]: yes, because there is still unsafe casting.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT2779-1

**Pull request merged**